### PR TITLE
Updated to micronaut version of libraries

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: ftgo-order-history-service/src/main/java/net/chrisrichardson/ftgo/cqrs/orderhistory/main/OrderHistoryServiceMain.java [hunk 1]
```diff
@@ -1,7 +1,8 @@
 package net.chrisrichardson.ftgo.cqrs.orderhistory.main;
 
-import io.eventuate.tram.consumer.common.TramConsumerCommonConfiguration;
-import io.eventuate.tram.consumer.kafka.EventuateTramKafkaMessageConsumerConfiguration;
+import io.eventuate.tram.consumer.common.spring.TramConsumerCommonConfiguration;
+import io.eventuate.tram.consumer.kafka.spring.EventuateTramKafkaMessageConsumerConfiguration;
+import io.eventuate.tram.messaging.common.spring.TramMessagingCommonAutoConfiguration;
```
Comment (cer) [inline ftgo-order-history-service/src/main/java/net/chrisrichardson/ftgo/cqrs/orderhistory/main/OrderHistoryServiceMain.java]: Services should not explicitly `@Import` `TramMessagingCommonAutoConfiguration`.
That configuration should be picked up AUTOmatically but `@EnableAutoConfiguration` because of https://github.com/eventuate-tram/eventuate-tram-core/blob/master/eventuate-tram-messaging/src/main/resources/META-INF/spring.factories

Comment (dartartem) [reply ftgo-order-history-service/src/main/java/net/chrisrichardson/ftgo/cqrs/orderhistory/main/OrderHistoryServiceMain.java]: I will fix

File: ftgo-delivery-service/src/integration-test/java/net/chrisrichardson/ftgo/deliveryservice/domain/CourierJpaTest.java [hunk 2]
```diff
@@ -22,6 +24,7 @@
   @Configuration
   @EnableJpaRepositories
   @EnableAutoConfiguration
+  @Import(EventuateCommonJdbcOperationsConfiguration.class)
```
Comment (cer) [inline ftgo-delivery-service/src/integration-test/java/net/chrisrichardson/ftgo/deliveryservice/domain/CourierJpaTest.java]: Why `@Import(EventuateCommonJdbcOperationsConfiguration.class)` .
This is internal and this test class doesn't need any of the beans it defines.

Comment (dartartem) [reply ftgo-delivery-service/src/integration-test/java/net/chrisrichardson/ftgo/deliveryservice/domain/CourierJpaTest.java]: I will check

File: ftgo-order-service/build.gradle [hunk 3]
```diff
@@ -78,7 +78,7 @@ dockerCompose {
 }
 
 integrationTest.dependsOn(integrationTestsComposeUp)
-
+componentTestsComposeUp.dependsOn(integrationTestsComposeUp)
```
Comment (cer) [inline ftgo-order-service/build.gradle]: Why is this necessary? The ftgo-order-service container `depends_on` mysql

Comment (dartartem) [reply ftgo-order-service/build.gradle]: It adds additional timeout to connect to mysql. Otherwise mysql is unavailable.

File: ftgo-order-service/src/main/java/net/chrisrichardson/ftgo/orderservice/domain/OrderServiceConfiguration.java [hunk 4]
```diff
@@ -38,8 +43,13 @@ public OrderService orderService(RestaurantRepository restaurantRepository, Orde
   }
 
   @Bean
-  public SagaManager<CreateOrderSagaState> createOrderSagaManager(CreateOrderSaga saga) {
-    return new SagaManagerImpl<>(saga);
+  public SagaManager<CreateOrderSagaState> createOrderSagaManager(CreateOrderSaga saga,
```
Comment (cer) [inline ftgo-order-service/src/main/java/net/chrisrichardson/ftgo/orderservice/domain/OrderServiceConfiguration.java]: `SagaManager` should not longer be used by applications.

Applications should use `SagaInstanceFactory` - see https://github.com/eventuate-tram/eventuate-tram-sagas/commit/77bd85b978f2e80dd3088a81d061a65f6d465668  and https://github.com/eventuate-tram/eventuate-tram-sagas-examples-customers-and-orders/commit/168cc54a1807abdfc5d0b99a1def9a39c1f36aaf

Comment (dartartem) [reply ftgo-order-service/src/main/java/net/chrisrichardson/ftgo/orderservice/domain/OrderServiceConfiguration.java]: Missed it. Sorry.

File: gradle.properties [hunk 5]
```diff
@@ -14,15 +14,14 @@ springDependencyManagementPluginVersion=1.0.3.RELEASE
...
-eventuateClientVersion=0.22.0.RC2
-eventuateLocalVersion=0.31.0.RC5
-
-eventuateTramVersion=0.22.0.RC7
-eventuateTramSagasVersion=0.12.0.RC6
-
-eventuateUtilVersion=0.2.0.RELEASE
-eventuateCommonVersion=0.8.3.RELEASE
-eventuateMessagingKafkaVersion=0.5.0.RELEASE
+eventuateClientVersion=0.22.0-SNAPSHOT
```
Comment (cer) [inline gradle.properties]: These can be replaced with replaced versions.

Comment (dartartem) [reply gradle.properties]: will do

File: ftgo-order-service/build.gradle (hunk 3 above)
Comment (cer) [reply ftgo-order-service/build.gradle]: Two things.

1. Why is that a problem? It becomes available eventually
2. I think explicitly including `mysql` as a startedService of componentTest configuration will cause Docker Compose plugin to wait.

Comment (dartartem) [reply ftgo-order-service/build.gradle]: 1) Order service fails to start because mysql is not available initially.
2) I will try

File: gradle.properties (hunk 5 above)
Comment (dartartem) [reply gradle.properties]: replaced, except kafka and sagas, it seems, they are not released.

File: ftgo-order-service/build.gradle (hunk 3 above)
Comment (dartartem) [reply ftgo-order-service/build.gradle]: fix from point 2 did not help

**Comment by cer** [comment]
Merged into branch https://github.com/microservices-patterns/ftgo-application/tree/wip-development-with-micronaut-libraries
